bigcode_eval/evaluator.py

```diff
@@ -52,41 +58,45 @@ def generate_text(self, task_name):
                 solutions = [[ref] for ref in references]
             return solutions, references
 
-        generations = parallel_generations(
+        generations = []  # list[list[str | None] | None]
+        if intermediate_generations:
+            generations = [gen for gen in intermediate_generations if gen]
+            n_tasks -= len(generations)
+        intermediate_save_generations_path = f"{os.path.splitext(self.args.save_generations_path)[0]}_{task_name}_intermediate.json"
+        curr_sample_idx = len(generations)
+
+        new_generations = parallel_generations(
             task,
             dataset,
             self.accelerator,
             self.model,
             self.tokenizer,
             n_tasks=n_tasks,
             args=self.args,
+            curr_sample_idx=curr_sample_idx,  # curr_sample_idx will added to limit_start to fix indexing
+            save_every_k_tasks=self.args.save_every_k_tasks,
+            intermediate_save_generations_path=intermediate_save_generations_path,
```

Comment:
It seems to me that the generations loaded from a previous run (`generations`) will not be saved in the new intermediate file.
So if the job was to be interrupted again, those previously loaded generations would be lost?

Reply:
good catch, I've added a fix for this, ptala